### ISSUE TITLE
Iteration Plan for Dec 2019

### ISSUE THREAD

**Issue by scarlett2018** [microsoft/nni#1794]
**Feature freeze date**: 16 Dec, 2019
**Code freeze date**: 10:30AM 24 Dec, 2019
**Release Manager**: @leckie-chn  
**End Game Issue**: is tracking in #1872 

-----------------------------------------------
### Planned Features
- [x] NFS support (PAI - Dian Wang) (#1635 ) –  #1848 **in review** @SparkSnail    
- [x] Kubeflow update adoption (investigation) @SparkSnail #1814 (resolving conflict)
- [x] Proxyless NAS @QuanluZhang #1863 

#### ~TensorFlow 2.0 support @liuzhe-lz  (PR eta. Fri.)~
- [ ] tensorflow 2.0 (tf.keras) support for NNI NAS interface (#1737 )  (debugging, eta. Fri.)
- [ ] tensorflow 2.0 support for ~static~ dynamic graph model compression   (depend on #1737 )
- [ ] tensorflow 2.0 test and example  (code 2d)

#### more NAS algorithm support @ultmaster 
- [x] SPOS (searching done, debugging for training, splitting PR into 2) #1849 #1865 
- [ ] ~C-DARTS (to be synced with partner)~  
- [ ] AutoNLP (to be synced author) 
     - [ ] search algo
- [ ] ~AutoDeepLab (TBD)~

#### more compression algorithm support @Cjkkkk @tanglang96 
- [x] support distill training mode @tanglang96 (#1819 )
- [x] Pruning algorithms: L2 pruning, FMreconstruction channel pruning, network trimming @tanglang96 #1867 
- [x] Quantization algorithms: @Cjkkkk #1832 
       - [ ] separate PR for doc 
       - [x] BNN #1854  
       * ~TTQ (for next release)~
       * ~Trained quantization and weight sharing (TBD)~

#### Pass eslint ( #882  )
- [X] eslint rules investigation ( #1825 ) @chicm-ms 
- [X] nni manager eslint: 
  - [X] Eslint config and auto fix  ( #1831  ) @chicm-ms
  - [X] manual fix ( #1833  ) @chicm-ms
  - [X] refactor to fix eslint ( #1835 ) @chicm-ms
  - [X] enable eslint in pipeline ( #1845 ) @chicm-ms 
- [x] Webui eslint
  - [x] Eslint config and fix .ts files ( #1847 ) @chicm-ms  reviewers: @liuzhe-lz  @ultmaster 
  - [x]  fix WebUI *.tsx files eslint and enable @lvybriage #1864 pending @liuzhe-lz review

#### Other Features
- [ ] ~tests for supported NAS algorithms (#1598 ) @ultmaster (TBD)~
- [ ] ~cancel trials in UNKNOWN state (#1735 ) @chicm-ms (pending, code 1d)~
- [ ] ~fabric framework adoption @lvybriage (coding, eta. 20th, Dec)~
- [ ] ~BUG - correctly support `NaN` in metric data, JSON compliant (#1589 ) @liuzhe-lz (code 1d) @chicm-ms to help profiling.~ 

- [x] correctly recognize builtin tuner and customized tuner @chicm-ms  #1862  (testing on `dev-refactor-dispatcher-cmdline`)
- [ ] logging in dispatcher_base @chicm-ms (bug fix after code freeze)
- [x] nnictl command auto complete @liuzhe-lz 
- [x] refactor tuner working directory (doc work) @chicm-ms  
- [x] properly catch tuner's error @liuzhe-lz 
- [x] Fix local system as remote machine issue #1852 @chicm-ms PR #1856  submitted

### Documentation
- [X] documentation structure refactor @scarlett2018 
   - [x] Readme update #1860 
- [ ] customized homepage for sphinx doc build @leckie-chn 
- [ ] write doc for IPC interface (#1741 )  @liuzhe-lz 
- [x] update the overview page to explain HPO, NAS, Compression, Feature Engineering @QuanluZhang (#1821 )
- [x] add some FAQs about the common network issue that could cause the start failure of restserver (#1223 ) @ultmaster @QuanluZhang (#1823 ) resolving conflict
- [x] document improvement of remote mode training service @ultmaster (#1823 )
- [ ] improvement of document structure @QuanluZhang @leckie-chn @scarlett2018  (To be reviewed)

### Small changes
- [ ] upgrade to the new cuda version for our maintained docker image (#1748, pending until before release ) @liuzhe-lz 
- [ ] ~remove hardcoded call to `/usr/bin/python3` in `gpu_metrics_collector.sh` (#1712 ) @liuzhe-lz~
- [x] de-duplicate trial configuration in smac tuner (#1364 ) @QuanluZhang #1840 

### Internal feature
- [ ] Philly Training Services @SparkSnail  (PR Fri.)
